HEN_HOUSE/egs++/geometry/egs_cd_geometry/egs_cd_geometry.h

```diff
@@ -565,11 +565,15 @@ class EGS_CDGEOMETRY_EXPORT EGS_CDGeometry : public EGS_BaseGeometry {
                 }
                 // Skip for concave geometries: particles can reenter the CD geometry after leaving.
                 // Check 2. below will catch these cases.
-                else if ((ixnew < 0 && tb <= boundaryTolerance) && (bg->isConvex()  &&
-                         (g[ibase] && g[ibase]->isConvex()))) {
-                    // (a) is true
-                    return ixnew;
-                }
+                // This follow has been commented out because it fails in cases
+                // where the inscribe geometries are themselves convex but
+                // form a concave shape. Instead, just continue to do more checks.
+//                 else if ((ixnew < 0 && tb <= boundaryTolerance) && (bg->isConvex()  &&
+//                          (g[ibase] && g[ibase]->isConvex()))) {
+//                     // (a) is true
+//                     return ixnew;
+//                 }
```

Comment:
Are you sure this is the right solution? I'm fairly sure you need this specific check. Look what happens in egs_view with this pull request:

![this does not look good](https://user-images.githubusercontent.com/1675666/42667439-702b383e-8600-11e8-9b3c-a343e6d0965e.png)

This is a simple conestack with mismatched radius. If you make the smaller one transparent in egs_view, it never sees the larger one.

Reply:
Oh, I think I goofed! My conestack was the only geometry in the simulation, so when a particle exits the smaller cylinder, it leaves the geometry, which is why egs_view can't see anything behind it, I think. If you inscribe the conestack in a box, then it looks fine. My bad!

Reply:
Correct, that's normal behaviour. Thanks for testing!